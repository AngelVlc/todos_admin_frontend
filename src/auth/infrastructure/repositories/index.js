export * from './RefreshTokensRepository';
export * from './UsersRepository';
