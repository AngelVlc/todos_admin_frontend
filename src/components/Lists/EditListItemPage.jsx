import React, { useEffect, useState } from 'react';
import { Link, useParams } from 'react-router-dom';
import { ListItemForm } from './ListItemForm';
import axios from 'axios';

export const EditListItemPage = () => {
    let { listId, itemId } = useParams();

    const [pageState, setPageState] = useState({title: ''});

    useEffect(() => {
        const getExistingItem = async () => {
            const res = await axios.get(`lists/${listId}/items/${itemId}`)
            const item = res.data

            setPageState({
                title: item.title,
                description: item.description
            });
        }
        getExistingItem()
    }, [listId, itemId]);

    return (
        <div className="container">
            <h3 className="title">{`Edit item '${pageState.title}'`}</h3>
            <nav className="breadcrumb" aria-label="breadcrumbs">
                <ul>
                    <li><Link to={`/`}>Home</Link></li>
                    <li><Link to={`/lists`}>Lists</Link></li>
                    <li><Link to={`/lists/${listId}/edit`}>List</Link></li>
                    <li className="is-active"><Link aria-current="page" to={`/lists/${listId}/items/${itemId}/edit`}>{pageState.isNew ? 'new' : pageState.title}</Link></li>
                </ul>
            </nav>
            <ListItemForm listId={listId} itemId={itemId} title={pageState.title} description={pageState.description} isNew={false} submintBtnText='SAVE' submitUrl={`lists/${listId}/items/${itemId}`}/>
        </div>
    )
}
