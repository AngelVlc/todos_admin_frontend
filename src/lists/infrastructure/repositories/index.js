export * from "./ListsRepository";
export * from "./ListItemsRepository";
