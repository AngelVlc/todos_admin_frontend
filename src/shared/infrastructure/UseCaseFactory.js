export class UseCaseFactory {
  get(useCase) {
    return useCase.create();
  }
}
