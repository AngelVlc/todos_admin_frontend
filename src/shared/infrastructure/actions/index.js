export * from './requests';
export * from './login';
