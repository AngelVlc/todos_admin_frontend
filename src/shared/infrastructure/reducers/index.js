export * from './loginReducer';
export * from './requestsReducer';
