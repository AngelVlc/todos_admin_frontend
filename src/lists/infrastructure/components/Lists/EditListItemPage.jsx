import React, { useEffect, useState, useContext, useCallback } from "react";
import { useParams } from "react-router-dom";
import { ListItemForm } from "./ListItemForm";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { GetListItemByIdUseCase } from "../../../application/listItems";
import { Breadcrumb } from "../../../../shared/infrastructure/components/Breadcrumb/Breadcrumb";

export const EditListItemPage = () => {
  let { listId, itemId } = useParams();
  const { useCaseFactory } = useContext(AppContext);
  const [pageState, setPageState] = useState({ title: "" });

  const getExistingItem = useCallback(async () => {
    const getListItemByIdUseCase = useCaseFactory.get(GetListItemByIdUseCase);
    const data = await getListItemByIdUseCase.execute(listId, itemId);

    setPageState(data);
  }, [listId, itemId, useCaseFactory]);

  useEffect(() => {
    getExistingItem();
  }, [listId, itemId, getExistingItem]);

  return (
    <div className="container">
      <Breadcrumb
        items={[
          { url: "/lists", text: "Lists" },
          { url: `/lists/${listId}`, text: "List" },
          {
            url: `/lists/${listId}/items/${itemId}/edit`,
            text: pageState.title,
          },
        ]}
      />
      <h3 className="title">{`Edit item '${pageState.title}'`}</h3>
      <ListItemForm listItem={pageState} />
    </div>
  );
};
