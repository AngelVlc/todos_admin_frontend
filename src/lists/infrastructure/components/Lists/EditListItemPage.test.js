import { render, cleanup } from '@testing-library/react';
import { EditListItemPage } from './EditListItemPage';
import { AppContext } from  '../../../../shared/infrastructure/contexts';
import axios from 'axios';
import { MemoryRouter, Route } from 'react-router-dom';
import { act } from 'react-dom/test-utils';

jest.mock('axios');
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        push: mockHistoryPush
    })
}));

const renderWithContextAndRouterForExistingItem = () => {
    axios.get.mockResolvedValue(
        {
            data:
            {
                id: 2,
                title: 'item title',
                description: 'item description'
            }
        }
    );
    const context = { auth: { info: {} } };
    return {
        ...render(
            <AppContext.Provider value={context}>
                <MemoryRouter initialEntries={[`/lists/2/items/5/edit`]}>
                    <Route path="/lists/:listId/items/:itemId/edit">
                        <EditListItemPage/>
                    </Route>
                </MemoryRouter>
            </AppContext.Provider>)
    }
}

afterEach(cleanup)

it('should match the snapshot for an existing item', async () => {
    let fragment;
    await act(async () => {
        const { asFragment } = renderWithContextAndRouterForExistingItem();
        fragment = asFragment;
    });
    expect(fragment()).toMatchSnapshot();
});
