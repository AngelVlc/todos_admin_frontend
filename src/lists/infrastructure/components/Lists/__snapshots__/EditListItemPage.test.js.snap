// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for an existing item 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <nav
        aria-label="breadcrumbs"
        class="breadcrumb"
      >
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/lists"
            >
              Lists
            </a>
          </li>
          <li>
            <a
              href="/lists/2"
            >
              List
            </a>
          </li>
          <li
            class="is-active"
          >
            <a
              aria-current="page"
              href="/lists/2/items/5/edit"
            >
              item title
            </a>
          </li>
        </ul>
      </nav>
      <h3
        class="title"
      >
        Edit item 'item title'
      </h3>
      <form
        action="#"
      >
        <div
          class="field"
        >
          <label
            class="label"
            for="title"
          >
            Title
          </label>
          <div
            class="control"
          >
            <input
              class="input"
              data-testid="title"
              name="title"
              value="item title"
            />
          </div>
          <p
            class="help is-danger"
            data-testid="titleErrors"
          />
        </div>
        <div
          class="field"
        >
          <label
            class="label"
            for="description"
          >
            Description
          </label>
          <div
            class="control"
          >
            <textarea
              class="textarea"
              data-testid="description"
              name="description"
            >
              item description
            </textarea>
          </div>
          <p
            class="help is-danger"
            data-testid="descriptionErrors"
          />
        </div>
        <div
          class="field is-grouped"
        >
          <div
            class="control"
          >
            <button
              class="button"
              data-testid="submit"
              type="submit"
            >
              SAVE
            </button>
          </div>
          <div
            class="control"
          >
            <button
              class="button"
              data-testid="cancel"
              type="button"
            >
              CANCEL
            </button>
          </div>
          <div
            class="control"
          >
            <button
              class="button is-danger"
              data-testid="delete"
              type="button"
            >
              DELETE
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
